.github/actions/backup-restore-test/action.yml

```diff
@@ -0,0 +1,120 @@
+name: Backup Restore Test
+
+inputs:
+  service:
+    description: Service to test
+    required: true
+  kind:
+    description: Kind of test
+    required: true
+  backup-args:
+    description: Arguments to pass for backup
+    required: false
+    default: ""
+  restore-args:
+    description: Arguments to pass for restore
+    required: false
+    default: ""
+  test-folder:
+    description: Folder to use for testing
+    required: true
+  base-backup:
+    description: Base backup to use for testing
+    required: false
+
+outputs:
+  backup-id:
+    value: ${{ steps.backup.outputs.result }}
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+    #TODO(meain): Do we need cache here?
```

Comment:
Yes, please.  Else we run into the possibility of re-downloading kiota libraries.

Reply:
![2023-05-15-20-59-32](https://github.com/alcionai/corso/assets/14259816/76376c61-436c-4b70-b989-812290ef50c0)

I was getting a lot of errors when trying to restore cache 🤔 . The job does succeed though.

Reply:
I'm gonna merge it in for now to get it going as the job still goes through. We can consider dropping it later in case this is not required.